src/pages/docs/guides/chat/build-livestream.mdx

```diff
@@ -0,0 +1,240 @@
+
+PLACEHOLDER: Diagram comparing single-room vs. multi-room architectures
+
+In making this decision, there are some **key considerations**:
+
+- Should everyone see the same messages, or do you need to segment by language, region, friend group or topic?
+  - If you're running livestreams over multiple languages or geographical regions, you may want to consider separating users similarly so that everybody understands the chat.
+  - Alternatively, you may want to make use of translation services, that can translate messages on the fly to the language of the user.
+- Do you want an environment where everyone has time to read every message, or the chat to go wild with hype when something exciting happens?
+  - At anything more than 5-10 messages per second, the chat becomes unreadable for the audience.
+  - **Sharding** is one approach to this problem - separating users into smaller groups so that the chat is more manageable. This can be done randomly, or using some predefined scheme, e.g. language, geographical region.
+  - You can also simply _drop_  a proportion of messages on the client to keep it readable. If you're not worried about everyone seeing everything, this is a simple approach.
+  - Of course, you can also rate-limit how often users send messages. This controls the rate of messages, but can negatively impact the user experience if they're not able to respond when they want to.
+- Do you need to support both public and private chat? If you do, you might want one big room for public chat and a separate room for private messages.
+
+Ably gives you the flexibility to design for your users, not your infrastructure.
+
+## Message throughput and rate management
+
+Livestream chat is defined by high message rates and unpredictable spikes. Ably is engineered to handle these loads without degradation.
+
+- **Proven at scale:** Ably delivers over 500 million messages per day for customers, with individual rooms supporting over 1 million concurrent users.
+- **Server-side batching:** Reduce costs and network overhead by grouping messages before delivery. For example, batching messages over a 500ms interval can reduce message costs by up to 8x in high-traffic streams.
+- **Rate limiting:** Control the flow of messages per user to maintain readability and prevent spam. Choose between per-user rate limits, global throttling, or adaptive algorithms.
+
+CODE HERE: Example of implementing per-user rate limiting logic
+
+PLACEHOLDER: Visual showing message spikes and how batching smooths delivery/costs
+
+**Decision point:**
+- What is the maximum message rate you want users to experience?
+- How will you balance real-time delivery with cost efficiency?
```

Comment:
I'm not sure if this is a "how will you" type question. Is this not more relevant when discussing server-side batching below?